src/ore/src/tracing.rs

```diff
@@ -316,12 +317,31 @@ where
             .install_batch(opentelemetry::runtime::Tokio)
             .unwrap();
         let (filter, filter_handle) = reload::Layer::new(if otel_config.start_enabled {
-            otel_config.filter
+            let targets = Targets::default()
+                // By default we turn off tracing from the following crates, because they
+                // have long-lived Spans, which OpenTelemetry does not handle well. We
+                // specifically apply the otel_config _after_ these defaults, to allow the
+                // otel_config to override this setting, if it's desired.
+                //
+                // Note: users should feel free to add more crates here if we find more
+                // with long lived Spans.
+                .with_targets([("h2", LevelFilter::OFF), ("hyper", LevelFilter::OFF)]);
+
+            // Apply the user provided config
+            match otel_config.filter.default_level() {
+                Some(defaul_level) => targets.with_default(defaul_level),
+                _ => targets,
+            }
+            .with_targets(otel_config.filter)
         } else {
             // The default `Targets` has everything disabled.
             Targets::default()
         });
         let layer = tracing_opentelemetry::layer()
+            // OpenTelemetry does not handle long-lived Spans well, and they end up continuously
+            // eating memory until OOM. So we set a max number of events that are allowed to be
+            // logged to a Span, once this max is passed, old events will get dropped
+            .max_events_per_span(256)
             .with_tracer(tracer)
             .with_filter(filter);
         let reloader = Arc::new(move |targets| Ok(filter_handle.reload(targets)?));
```

Comment:
Ideally the `h2` and `hyper` filters need to be added here, if they aren't explicitly set by the filter being reloaded!

Reply:
Updated to do this!

The code to update this is def duplicated with the code above, I tried de-duping it into a closure, but it didn't look great so I stuck with the slightly duped code. If you want me to change this let me know :)

Reply:
yeah, no worries, duplication can be good sometimes!